api/v1alpha1/addon_types.go

```diff
@@ -46,6 +48,10 @@ func (Addons) VariableSchema() clusterv1.VariableSchema {
 			Description: "Cluster configuration",
 			Type:        "object",
 			Properties: map[string]clusterv1.JSONSchemaProps{
+				"helmChartRepository": {
+					Pattern:     patterns.HostAndOptionalPort,
+					Description: "Optional OCI registry used to pull helm charts for adons",
```

Comment:
Is this just an OCI registry or could it also be https helm repository somewhere?

Reply:
You could expand this description and add example too via `Example` property.

Reply:
I'm not sure what the difference between helm repository and OCI registry is. That entity would need to host all of the charts though. because the repository field for each addon gets set to this value.

Reply:
Helm repositories used to be supported only accessible via https. OCi support was later introduced meaning charts could be stored and referenced from OCI registries. The only difference is the scheme in the repository URL, using oci:// instead of https://. I think we should support both here.

Reply:
I don't know if all the users are going to be able to push helm charts to a single helm repository im also not sure how to validate it?

Reply:
This looks like it's only supporting pulling from one helm repository though right for all Helm addons?

Reply:
yeah, exactly the idea is that this source will have all of the helm charts the user needs